### ISSUE TITLE
ts.NodeFlags.Namespace should be set for nested namespaces declared with . syntax

### ISSUE THREAD

**Issue by JKillian** [Microsoft/TypeScript#4436]
If a nested namespace is declared using the dot syntax, it is not flagged as a namespace in its `ts.Node.flags` field:

```typescript
namespace A.B { }
// moduleDeclarationNodeA.flags === ts.NodeFlags.Namespace
// moduleDeclarationNodeB.flags === ts.NodeFlags.Export
// Should be: moduleDeclarationNodeB.flags === ts.NodeFlags.Export | ts.NodeFlags.Namespace
```

This is related [to a tsLint issue](https://github.com/palantir/tslint/issues/600) where we enforce using `namespace` instead of `module` to avoid confusion with external modules.

**Comment by JKillian** [comment]
I could possibly do a PR to fix this if you agree with the issue (seems like [this code would have to be tweaked](https://github.com/Microsoft/TypeScript/blob/ecaf16df4b88d9f0e66daef6afa18cd983d9ecc5/src/compiler/parser.ts#L4942-L4975)) but probably simpler to just fix it on your end

**Comment by mhegazy** [comment]
@JKillian feel free to send a PR for this. you just need to pass the NodeFlags.Namespace in the subsequent invocations of `parseModuleOrNamespaceDeclaration` from `parseModuleDeclaration`

**Comment by JKillian** [comment]
@mhegazy Will do, that's what I was thinking as well

**Comment by JKillian** [comment]
@mhegazy Could you give me a little guidance on tests for this issue? I'm not sure on the proper method to test this.

**Comment by DanielRosenwasser** [comment]
@CyrusNajmabadi can this affect incremental scenarios?

It's odd since the node flag is only used to indicate that if the keyword was there, a .d.ts file should emit with `namespace` instead of `module`. TSLint could check the parent pointers to see if it makes sense to even make the check.

**Comment by JKillian** [comment]
There are definitely work-arounds as far as TSLint is concerned, so it's not a critical issue. I'll hold off on any PRs for now as I don't know the impact of the change.

**Comment by vmmitev** [comment]
@DanielRosenwasser If node flags are set only if the keyword is part of the node, then what value can we expect for the `flags` property on nodes that are nested with `.` syntax?

For example:
```typescript
namespace A.B {}
module C.D {}
```
During debugging my observations were that nodes for `B` and `D` had a `flags` property equal to 1, which to my understanding is the `ts.NodeFlags.Export` flag. `C`'s on the other hand is 0, unless used with `declare` in which case the value turns to 2. And of course `A` has a value of `ts.NodeFlags.Namespace = 32768`.

If all of this checks out from the compiler's point of view, then checking parent pointers, as you suggested, becomes the only option.

**Comment by mhegazy** [comment]
We propagate the other flags. I do not see a problem setting the Namespcae flag on all nodes. we do not currently use them in the compiler, but seem more consistent this way.

**Comment by CyrusNajmabadi** [comment]
I don't see a problem here from the perspective of incremental parsing.
